### PR TITLE
11 adding delete monitor functionality

### LINKED ISSUE #11: Adding delete monitor functionality
Client side:
Adding delete button on each monitor which calls DELETE /monitor:id

Server side:
Completing the DELETE route. Currently, it only deletes the monitor from the database and does not stop the retrieveMonitor interval set. This leads to zombie code until the server restarts.

### PR DESCRIPTION
**Pull request by mjw324**

### CODE REVIEW

File: client/eSentry/src/app/components/monitor-list/monitor-list.component.html [hunk 1]
```diff
@@ -23,7 +24,7 @@ <h1>Monitor List</h1>
       </td>
       <td>
         <ng-container>
-          <ng-template [ngIf]="monitor.condition_new">
+          <ng-template [ngIf]="monitor.conditio_nnew">
```
Comment (paolobarto) [inline client/eSentry/src/app/components/monitor-list/monitor-list.component.html]: Is this a forced name?

File: client/eSentry/src/app/components/monitor-list/monitor-list.component.html [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <h1>Monitor List</h1>
-<p-table [value]="monitors">
+<p-table [value]="(monitors$ | async) ?? []">
```
Comment (paolobarto) [inline client/eSentry/src/app/components/monitor-list/monitor-list.component.html:2]: Usage of asnyc pipes is badass for this. Definitely overall removes a layer of logic between service and outcome. I will implement this as much as I can on my end!

File: client/eSentry/src/app/services/monitor.service.ts [hunk 3]
```diff
@@ -12,33 +13,42 @@ import { MessageService } from 'primeng/api';
...
-  addMonitor(request: MonitorRequest, userid: string) {
-    this.http
-      .post<Monitor>(environment.url + '/monitors', request)
-      .subscribe((monitor) => {
-        this.fetchMonitors(userid);
-      });
+  addMonitor(request: MonitorRequest, userid: string): Observable<Monitor> {
+    return this.http.post<Monitor>(environment.url + '/monitors', request)
+      .pipe(
+        tap(() => this.fetchMonitors(userid)),
```
Comment (paolobarto) [inline client/eSentry/src/app/services/monitor.service.ts:37]: This looks good, using tap to only consume 1 return is awesome for this case.

File: client/eSentry/src/app/models/monitor.model.ts [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
-    maxprice?: number;
-    excludekeywords?: string;
-    conditionnew?: boolean;
-    conditionopenbox?: boolean;
-    conditionused?: boolean;
-}
+    min_price?: number;
+    max_price?: number;
+    exclude_keywords?: string;
+    condition_new?: boolean;
```
Comment (paolobarto) [inline client/eSentry/src/app/models/monitor.model.ts:9]: Potential Bug: previous comment had condition_nnew

**Review by paolobarto — COMMENTED** [review]
check `condition_nnew` other than that, looks awesome with modern rxjs stuff. The new syntax you use took me a little bit to understand on my own (near when I stopped working), so just make sure you "wield" these tools effectively.

File: client/eSentry/src/app/components/monitor-list/monitor-list.component.html (hunk 1 above)
Comment (mjw324) [reply client/eSentry/src/app/components/monitor-list/monitor-list.component.html]: Yeah that was an accident, I will change it here

**Review by paolobarto — APPROVED** [review]

**Pull request merged**